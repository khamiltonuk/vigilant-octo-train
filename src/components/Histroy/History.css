ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

li {
  float: left;
  background: #d1df39;
  padding: 10px;
  margin-right: 5px;
  margin-bottom: 5px;
}
