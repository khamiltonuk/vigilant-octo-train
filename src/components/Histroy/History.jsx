import React from "react";

import "./History.css";

export default function History(props) {
  const { removeValue, values } = props;
  return (
    <ul>
      {values.map(el => {
        return (
          <li>
            {el} <button onClick={removeValue}>x</button>
          </li>
        );
      })}
    </ul>
  );
}
