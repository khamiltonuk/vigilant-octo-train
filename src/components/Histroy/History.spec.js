import React from "react";
import { shallow, simulate } from "enzyme";
import History from "./History";
import toJson from "enzyme-to-json";

describe("History", () => {
  test("should return a list of four items when given 4 values", () => {
    const props = {
      values: [25, 35, 535, 53]
    };
    const wrapper = shallow(<History {...props} />);
    expect(wrapper.find("li").length).toBe(4);
  });

  test("should match the snapshot", () => {
    const props = {
      values: [25, 35, 535, 53]
    };
    const wrapper = shallow(<History {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test.skip("should behave...", () => {
    const props = {
      values: [25, 35, 535, 53],
      removeValue: () => true
    };
    const wrapper = shallow(<History {...props} />);
    const button = wrapper.find("button").first();
    button.simulate("click");
    expect(wrapper.find("li").length).toBe(3);
  });
});
