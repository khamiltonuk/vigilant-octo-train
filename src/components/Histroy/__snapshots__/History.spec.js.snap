// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History should behave... 1`] = `
<ul
  className="history group"
>
  <FlipMovePropConverter
    delay={0}
    disableAllAnimations={false}
    duration={250}
    easing="ease-out"
    enterAnimation="elevator"
    getPosition={[Function]}
    leaveAnimation="elevator"
    maintainContainerHeight={false}
    staggerDelayBy={0}
    staggerDurationBy={0}
    typeName="div"
    verticalAlignment="top"
  >
    <li
      key="0"
    >
      25
       
      <button
        onClick={[Function]}
      >
        ✖
      </button>
    </li>
    <li
      key="1"
    >
      35
       
      <button
        onClick={[Function]}
      >
        ✖
      </button>
    </li>
    <li
      key="2"
    >
      535
       
      <button
        onClick={[Function]}
      >
        ✖
      </button>
    </li>
    <li
      key="3"
    >
      53
       
      <button
        onClick={[Function]}
      >
        ✖
      </button>
    </li>
  </FlipMovePropConverter>
</ul>
`;

exports[`History should match the snapshot 1`] = `
<ul
  className="history group"
>
  <FlipMovePropConverter
    delay={0}
    disableAllAnimations={false}
    duration={250}
    easing="ease-out"
    enterAnimation="elevator"
    getPosition={[Function]}
    leaveAnimation="elevator"
    maintainContainerHeight={false}
    staggerDelayBy={0}
    staggerDurationBy={0}
    typeName="div"
    verticalAlignment="top"
  >
    <li
      key="0"
    >
      25
       
      <button
        onClick={[Function]}
      >
        ✖
      </button>
    </li>
    <li
      key="1"
    >
      35
       
      <button
        onClick={[Function]}
      >
        ✖
      </button>
    </li>
    <li
      key="2"
    >
      535
       
      <button
        onClick={[Function]}
      >
        ✖
      </button>
    </li>
    <li
      key="3"
    >
      53
       
      <button
        onClick={[Function]}
      >
        ✖
      </button>
    </li>
  </FlipMovePropConverter>
</ul>
`;
